.github/workflows/ci.yml

```diff
@@ -363,3 +368,31 @@ jobs:
         run: ./scripts/ci/ci_prepare_image_on_ci.sh
       - name: "Push CI image ${{ matrix.python-version }}"
         run: ./scripts/ci/ci_push_ci_image.sh
+
+  tag-repo-nightly:
+    timeout-minutes: 60
+    name: "Tag repo nightly"
+    runs-on: ubuntu-latest
+    needs:
+      - tests-sqlite
+      - tests-postgres
+      - tests-mysql
+      - requirements
+      - prepare-backport-packages
+      - build-prod-image
+      - docs
+    if: github.event_name == 'schedule'
+    steps:
+      - uses: actions/checkout@master
+      - name: "Free space"
+        run: ./scripts/ci/ci_free_space_on_ci.sh
+      - name: "Build CI image"
+        run: ./scripts/ci/ci_prepare_image_on_ci.sh
+      - name: "Tag commit"
+        run: |
+          git tag -f nightly-master HEAD
+      - name: Push tags
+        uses: ad-m/github-push-action@master
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          tags: true
```

Comment:
Does this push to Master directly ??? If so, we shouldn't do that instead create a PR like the Dependency Bot

Reply:
It only pushes nightly tag  - nothing else. I am adding an extra protection to make sure that this always  a good branch. But the whole idea is that this job pushes "nighty-master" tag after all the tests in nightly cron job are fine. I am also adding some comprehensive description of how the CI jobs work as different jobs have different purpose. Bear with me :)

Reply:
Sure

Reply:
There it goes @kaxil (and others) . Complete with description of the CI infrastructure we use and even a nice image showing the infrastructure.